DiamondLightSource/SynchWeb
# LIMS-1257: Allow population of source fields

**JIRA ticket**: [LIMS-1257](https://jira.diamond.ac.uk/browse/LIMS-1257)

**Summary**:

Karl has added a source column to the BLSample, Container, Dewar and Shipping tables. By default it gets populated with the user name of the current database user, but we should allow something more descriptive in there, for eg IceBear, future web-based CSV uploader etc

**Changes**:
- Add a 'SOURCE' parameter allowed in Shipment.php and Sample.php
- Default it to null
- Put it in insert statements, but if it is null, use the current database user as that is better than nothing

**To test**:
- Create a shipment in Synchweb with at least one dewar, container and puck. Check the 4 source columns are all populated with a non-null user name such as `ispyb_web@%` or `synchweb_dev@%` (actual username depends on your dev setup)
- Repeat adding another dewar to the shipment, as that uses a different function to dewars created with the shipment
- Use Firefox's "Edit and Resend" functionality to repeat the creation of each, adding a `SOURCE` parameter each time, check this goes into the database. Dewars created at the same time as the shipment will have the same source.
- Alternatively, use the Selenium API test (https://github.com/DiamondLightSource/SynchWeb-Selenium/blob/main/tests/mx/test_mx_shipping_api.py) to create Shipments/Dewars/Pucks/Samples, and add a source value when doing so. Check each goes into the database.
